util/ksonnet/ksonnet.go

```diff
@@ -56,66 +92,39 @@ func KsonnetVersion() (string, error) {
 }
 
 type ksonnetApp struct {
-	app  app.App
-	spec app.Spec
+	rootDir string
 }
 
 // NewKsonnetApp tries to create a new wrapper to run commands on the `ks` command-line tool.
 func NewKsonnetApp(path string) (KsonnetApp, error) {
-	ksApp := ksonnetApp{}
-	a, err := app.Load(afero.NewOsFs(), path, false)
-	if err != nil {
-		return nil, err
-	}
-	ksApp.app = a
-
-	var spec app.Spec
-	err = config.UnmarshalLocalFile(filepath.Join(a.Root(), "app.yaml"), &spec)
-	if err != nil {
+	ksApp := ksonnetApp{rootDir: path}
+	// ensure that the file exists
+	if _, err := ksApp.appYamlPath(); err != nil {
 		return nil, err
 	}
-	ksApp.spec = spec
 	return &ksApp, nil
 }
 
-func (k *ksonnetApp) ksCmd(args ...string) (string, error) {
-	cmd := exec.Command("ks", args...)
-	cmd.Dir = k.Root()
-
-	cmdStr := strings.Join(cmd.Args, " ")
-	log.Debug(cmdStr)
-	outBytes, err := cmd.Output()
-	if err != nil {
-		exErr, ok := err.(*exec.ExitError)
-		if !ok {
-			return "", err
-		}
-		errOutput := string(exErr.Stderr)
-		log.Errorf("`%s` failed: %s", cmdStr, errOutput)
-		return "", fmt.Errorf(strings.TrimSpace(errOutput))
+func (k *ksonnetApp) appYamlPath() (string, error) {
+	const appYamlName = "app.yaml"
+	p := filepath.Join(k.rootDir, appYamlName)
+	if _, err := os.Stat(p); err != nil {
+		return "", err
 	}
-	out := string(outBytes)
-	log.Debug(out)
-	return out, nil
+	return p, nil
 }
 
-func (k *ksonnetApp) Root() string {
-	return k.app.Root()
-}
-
-// App is the Ksonnet application
-func (k *ksonnetApp) App() app.App {
-	return k.app
+func (k *ksonnetApp) ksCmd(args ...string) (string, error) {
+	return ksCmd(k.rootDir, args...)
 }
 
-// Spec is the Ksonnet application spec
-func (k *ksonnetApp) Spec() *app.Spec {
-	return &k.spec
+func (k *ksonnetApp) Root() string {
+	return k.rootDir
 }
 
 // Show generates a concatenated list of Kubernetes manifests in the given environment.
 func (k *ksonnetApp) Show(environment string) ([]*unstructured.Unstructured, error) {
-	out, err := k.ksCmd("show", environment)
+	out, err := ksCmd(k.rootDir, "show", environment)
```

Comment:
It seems all of the ksonnet methods have been switching to use a new package level ksCmd which accepts the ksonnet root as the first argument. The point of having a method-level ksCmd was so that the callers didn't have to keep passing the root dir. I don't think the package level ksCmd is necessary if the only reason it was made package-level was to be able to run `ksonnet version` without an app, in which case the KsonnetVersion() function could have been implemented using a throwaway app. e.g.:

```
ksApp := ksonnetApp{}	
ksApp.ksCmd("version")
```